planner/optimize.go

```diff
@@ -199,8 +199,9 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 
 	// try to get Plan from the NonPrepared Plan Cache
 	if sctx.GetSessionVars().EnableNonPreparedPlanCache &&
-		isStmtNode &&
-		!useBinding { // TODO: support binding
+		isStmtNode && !useBinding && !sctx.GetSessionVars().StmtCtx.InExplainStmt {
```

Comment:
Please put this check `!sctx.GetSessionVars().StmtCtx.InExplainStmt` inside the function `getPlanFromNonPreparedPlanCache`.